src/Sarif/SarifResources.resx

```diff
@@ -201,4 +198,16 @@
   <data name="CppCheckMissingLocation" xml:space="preserve">
     <value>A CppCheck entry must have a nonzero number of locations.</value>
   </data>
+  <data name="InvalidState" xml:space="preserve">
+    <value>One or more invalid states were detected during serialization: {0}</value>
```

Comment:
I doubt an end user will understand this message. Is it saying that we tried to read a file whose contents were invalid? Is it saying that we called methods of the logging API in the wrong order? Something else? Whichever it is, let's say that.